nodes/test/test_pinecone_record_updates.py

```
@@ -0,0 +1,290 @@
+            for record_id in ids:
+                self.records.pop(record_id, None)
+
+    def update(self, record_id: str | None = None, set_metadata: dict | None = None, **kwargs: object) -> None:
+        self.update_calls += 1
+        if record_id is None:
+            record_id = kwargs.get('id')
+
+        if record_id in self.records and set_metadata:
+            self.records[record_id]['metadata'].update(set_metadata)
+
+    def _matches_filter(self, metadata: dict, filter_expr: dict | None) -> bool:
```

Ruff (0.15.6)

[warning] Too many return statements (8 > 6)

(PLR0911)